parsl/executors/high_throughput/interchange.py

```diff
@@ -481,7 +481,7 @@ def process_tasks_to_send(self, interesting_managers: Set[bytes]) -> None:
                         m['idle_since'] = None
                         logger.debug("Sent tasks: %s to manager %r", tids, manager_id)
                         # recompute real_capacity after sending tasks
-                        real_capacity = m['max_capacity'] - tasks_inflight
+                        real_capacity -= task_count
```

Comment:
thinking this through to understand it myelf.. and if rewriting this a bit more...

the point of this line is to feed into the `if` statement below, and nowhere else. the `if` statement is asking "did we schedule as many tasks as we wanted to (did will fill the original `real_capacity`?) or do we still have capacity (in which case the manager remains interesting). the actual numeric value of `real_capacity` is irrelevant here beyond whether it is positive or not.

This assertion should hold, though:
`assert real_capacity == m['max_capacity'] - len(m['tasks'])` which is the calculation that happens earlier at line 471.

Reply:
... and then maybe the line 486 debug message would be phrased as "*still* has free capacity"